pkg/reconciler/awscognitosource/adapter.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright (c) 2020 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awscognitosource
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"knative.dev/pkg/kmeta"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/metrics"
+	pkgreconciler "knative.dev/pkg/reconciler"
+
+	"github.com/triggermesh/aws-event-sources/pkg/apis/sources/v1alpha1"
+	"github.com/triggermesh/aws-event-sources/pkg/reconciler"
+	"github.com/triggermesh/aws-event-sources/pkg/reconciler/resource"
+)
+
+const adapterName = "awscognitosource"
+
+const (
+	identityPoolIdEnvVar     = "IDENTITY_POOL_ID"
+	awsRegionEnvVar          = "AWS_REGION"
+	awsAccessKeyIdEnvVar     = "AWS_ACCESS_KEY_ID"
+	awsSecretAccessKeyEnvVar = "AWS_SECRET_ACCESS_KEY"
+)
+
+// adapterConfig contains properties used to configure the source's adapter.
+// These are automatically populated by envconfig.
+type adapterConfig struct {
+	// Container image
+	Image string `default:"gcr.io/triggermesh/awscognito"`
+	// Logging configuration, serialized as JSON.
+	LoggingCfg string `ignored:"true"`
+	// Metrics (observability) configuration, serialized as JSON.
+	MetricsCfg string `ignored:"true"`
+}
+
+// reconcileAdapter reconciles the state of the source's adapter.
+func (r *Reconciler) reconcileAdapter(ctx context.Context,
+	src *v1alpha1.AWSCognitoSource) (*appsv1.Deployment, error) {
+
+	sinkRef := &src.Spec.Sink.Ref
```

Comment:
wouldn't it panic if there is no `Sink` in source object? Asking because in knative sources it can be the case and currently `Sink` is being verified in the webhook: https://github.com/knative/eventing-contrib/pull/1120#discussion_r406812534

Reply:
You're absolutely right. My intention is to implement the essential parts of the validation in #65 before the feature branch gets merged to `master`, so I didn't bother validating too many things inside the reconciliation code because I would most likely delete it afterwards.

Reply:
Ok, makes sense